lib/lightning_web/live/run_live/index.ex

```diff
@@ -480,40 +479,30 @@ defmodule LightningWeb.RunLive.Index do
   end
 
   def handle_event("confirm-export", _params, socket) do
-    search_params = SearchParams.new(socket.assigns.filters)
-
-    case Projects.File.new(%{
-           type: :export,
-           status: :enqueued,
-           created_by: socket.assigns.current_user,
-           project: socket.assigns.project
-         })
-         |> Repo.insert() do
-      {:ok, project_file} ->
-        case ExportWorker.enqueue_export(
-               socket.assigns.project,
-               project_file,
-               search_params
-             ) do
-          :ok ->
-            {:noreply,
-             socket
-             |> assign(:show_export_modal, false)
-             |> put_flash(
-               :info,
-               "History export started successfully. You will be notified by email after completion."
-             )}
-
-          {:error, _reason} ->
-            Projects.File.mark_failed(project_file)
-            |> Repo.update!()
-
-            {:noreply,
-             socket
-             |> put_flash(:error, "Failed to start export. Please try again.")}
-        end
-
-      {:error, _changeset} ->
+    %{filters: filters, project: project, current_user: current_user} =
+      socket.assigns
+
+    search_params = SearchParams.new(filters)
+
+    case Invocation.export_workorders(project, current_user, search_params) do
+      {:ok, _} ->
+        {:noreply,
+         socket
+         |> assign(:show_export_modal, false)
+         |> put_flash(
+           :info,
+           "History export started successfully. You will be notified by email after completion."
+         )}
+
+      {:error, :export_job, _reason, %{project_file: project_file}} ->
+        Projects.File.mark_failed(project_file)
```

Comment:
@elias-ba I like what you did when you extracted the login into `Invocation.export_workorders`  but now I am wondering if calling `mark_failed` should not live in there rather? What do you think?

Reply:
I think it's a good idea. Updating the project file should be a responsibility of the presentation layer. I updated it.